pfio/cache/http_cache.py

```diff
@@ -33,36 +34,66 @@ def __setstate__(self, state):
     def is_forked(self):
         return self.pid != os.getpid()
 
-    def urlopen(self, method, url, redirect=True, **kw):
-        if self.is_forked or self.conn is None:
-            self.conn = urllib3.poolmanager.PoolManager(
-                retries=self.retries,
-                timeout=self.timeout
-            )
-            self.pid = os.getpid()
-        return self.conn.urlopen(method, url, redirect, **kw)
+    @contextlib.contextmanager
+    def get(self, host: str) -> \
+            Generator[http.client.HTTPConnection, None, None]:
+        if self.is_forked:
+            self.conn = collections.defaultdict(list)
+
+        conn: http.client.HTTPConnection
+        try:
+            conn = self.conn[host].pop()
+        except (IndexError, AssertionError):
+            conn = self._new_connection(host)
+
+        try:
+            yield conn
+        finally:
+            self.conn[host].append(conn)
+
+    def _new_connection(self, host: str) -> http.client.HTTPConnection:
+        return http.client.HTTPConnection(
+            host,
+            timeout=self.timeout
+        )
 
 
 CONNECTION_POOL: Optional[_ConnectionPool] = None
 
 
-def _get_connection_pool(retries: int, timeout: int) -> _ConnectionPool:
+def _get_connection_pool(timeout: int) -> _ConnectionPool:
     global CONNECTION_POOL
     if CONNECTION_POOL is None:
-        CONNECTION_POOL = _ConnectionPool(retries, timeout)
+        CONNECTION_POOL = _ConnectionPool(timeout)
     return CONNECTION_POOL
 
 
 class HTTPConnector(object):
     def __init__(self,
                  url: str,
                  bearer_token_path: Optional[str] = None,
-                 retries: int = 1,
                  timeout: int = 3):
-        if url.endswith("/"):
-            self.url = url
+        parsed = urllib.parse.urlparse(url)
```

Comment:
url parsing only happened in the constructor. previously it happened in get/put.